audit/internal/scenarios/scenarios_test.go

```diff
@@ -0,0 +1,106 @@
+package scenarios
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/microsoft/go-otel-audit/audit"
+	"github.com/microsoft/go-otel-audit/audit/base"
+	"github.com/microsoft/go-otel-audit/audit/conn"
+	"github.com/microsoft/go-otel-audit/audit/msgs"
+)
+
+func TestSlowListeningMDSD(t *testing.T) {
+	t.Parallel()
+
+	const auditLogQueueSize = 1
+
+	var validRecord = msgs.Record{
+		CallerIpAddress:            msgs.MustParseAddr("192.168.0.1"),
+		CallerIdentities:           map[msgs.CallerIdentityType][]msgs.CallerIdentityEntry{msgs.UPN: {{"user1@domain.com", "Description"}}},
+		OperationCategories:        []msgs.OperationCategory{msgs.UserManagement},
+		TargetResources:            map[string][]msgs.TargetResourceEntry{"ResourceType": {{"Name", "Cluster", "DataCenter", "Region"}}},
+		CallerAccessLevels:         []string{"Level1"},
+		OperationAccessLevel:       "AccessLevel",
+		OperationName:              "Operation",
+		OperationResultDescription: "ResultDescription",
+		CallerAgent:                "Agent",
+	}
+
+	id := uuid.New().String()
+	socketName := fmt.Sprintf("/tmp/mysocket-%s.sock", id)
+	defer os.Remove(socketName)
+
+	// Create a Unix domain socket and listen for incoming connections.
+	// However, for this, we are not going to listen at all.
+	socket, err := net.Listen("unix", socketName)
+	if err != nil {
+		panic(err)
+	}
+	defer socket.Close()
+
+	// Create a function that will create a new connection to the remote audit server.
+	// We use this function to create a new connection when the connection is broken.
+	cc := func() (conn.Audit, error) {
+		return conn.NewDomainSocket(conn.DSPath(socketName))
+	}
+
+	// Creates the smart client to the remote audit server.
+	// You should only create one of these, preferrably in main().
+	c, err := audit.New(cc, audit.WithAuditOptions(base.WithSettings(base.Settings{QueueSize: auditLogQueueSize})))
+	if err != nil {
+		panic(err)
+	}
+	//defer c.Close(context.Background())
+
+	go func() {
+		c, err := socket.Accept()
+		if err != nil {
+			panic(err)
+		}
+
+		b := make([]byte, 1024)
+		for {
+			c.Read(b)
+			time.Sleep(500 * time.Millisecond)
+		}
+	}()
+
+	for i := 0; i < 10000; i++ {
+		// Send a message to the remote audit server.
+		if err := c.Send(context.Background(), msgs.Msg{Type: msgs.ControlPlane, Record: validRecord}); err != nil {
+			log.Printf("msg(%d): %s}", i, err)
+		}
+	}
+
+	log.Println("Done")
+
+	time.Sleep(2 * time.Second)
```

Comment:
What's this wait for?

Reply:
I initially thought you were letting the test exit before all the messages were sent but I realize now that most sends (all but the first?) will just get ErrQueueFull...

Do you want some asserts to that effect? i.e. you expect more than 50% of the 10k sends to have hit ErrQueueFull?

Reply:
Probably something I had for something I took out.  Took it out.